File: cardinals.py
#!/usr/bin/python3
"""string of arrows, eaqch pointing to a different cardinal point
find the minimum number of arrows that must be rotated so they all point
in the same direction"""


def cardinal(points):
    """take the string, count how many are pointing in one direction"""
    if points is None:
        print("No points passed")
    else:
        point = []
        point[:] = points
        upcount = 0
        rcount = 0
        dcount = 0
        lcount = 0
        for arrow in range(len(point)):
            if point[arrow] == '^':
                upcount += 1
            elif point[arrow] == '>':
                rcount += 1
            elif point[arrow] == 'v':
                dcount += 1
            elif point[arrow] == '<':
                lcount += 1
        maxi = max([upcount, rcount, dcount, lcount])
        numb = len(point)
        if numb - maxi == 0:
            print("All arrows face the same direction")
        else:
            print(numb-maxi)


if __name__ == "__main__":
    cardinal('^vv<v')

File: cardinals.java
import java.util.Scanner;
import java.lang.Math;

public class cardinals {
    public static void main(String [] args){
        /*string of arrows, eaqch pointing to a different cardinal point
find the minimum number of arrows that must be rotated so they all point
in the same direction */
        Scanner myScanner = new Scanner(System.in);
        String points = myScanner.nextLine();
        myScanner.close();
        int up = 0, left = 0, down = 0, right = 0;
        for (int i = 0; i < points.length(); i++){
            char c = points.charAt(i);
            if (c == '<'){
                left++;
            }
            else if (c == '^'){
                up++;
            }
            else if (c == '>'){
                right++;
            }
            else if (c == 'v'){
                down++;
            }
            else{
                System.out.printf("Wrong input, '%s' not supported", c);
                return;
            }
        }
        int max = Math.max(Math.max(Math.max(down, right), up), left);
        int num = points.length();
        if (num-max == 0){
            System.out.println("All points to the same direction");
        }
        else{
            System.out.println(num-max);
        }
    }
}
